Source/qol/stash.h

```diff
@@ -18,45 +18,87 @@ class StashStruct {
 public:
 	using StashCell = uint16_t;
 	using StashGrid = std::array<std::array<StashCell, 10>, 10>;
-	static constexpr StashCell EmptyCell = -1;
-
-	void RemoveStashItem(StashCell iv);
-	std::map<unsigned, StashGrid> stashGrids;
-	std::vector<Item> stashList;
+	static constexpr StashCell EmptyCell = -1; // uint16_t(-1) or 65535
+	std::map<unsigned, StashGrid> stashGrids;  // the stash represented in pages containing a grid of itemIds
+	std::vector<Item> stashList;               // list of all items stored in the stash
 	int gold;
 	bool dirty = false;
 
+	/**
+	 * @brief Remove the passed item from the Stash, handling grid cleanup and index updates.
+	 * @param item pointer to item being removed.
+	 * @param itemPosition position of the item as it may not currently be (ex: autoequipped).
+	 * @param itemIndex 0-based index of the item as it exists in the stashList.
+	 */
+	void RemoveStashItem(Item *item, Point itemPosition, StashCell itemIndex);

```

Comment:
clang-tidy configuration is `methodCase=camelBase` for class methods, existing code still needs to be updated but new methods should follow the expected format.